virtcontainers/sandbox.go

```diff
@@ -952,6 +952,11 @@ func (s *Sandbox) startVM() error {
 	return s.agent.startSandbox(s)
 }
 
+// stopVM: stop the sandbox's VM
+func (s *Sandbox)stopVM() error {
+	return s.hypervisor.stopSandbox()
```

Comment:
We want to stop the sandbox with `s.agent.stopSandbox()` before we actually stop the VM. This way we mirror/rollback what has been done inside `startVM()`.

Reply:
Oh I've just seen your next commit and I can see you don't need `s.agent.stopSandbox()` since it actually removes the part `s.agent.startSandbox()` from `startVM()`.
This is fine, but please re-organize your code so that there is no inconsistent commit.

Reply:
Yes, you are right! I don't consider the case that if `s.agent.startSandbox()` is executed successfully, but `s.createContainers()` fails in `createSandboxFromConfig()` function, my code doesn't process this case well, I think I need to add `s.agent.stopSandbox()` before `s.stopVM()` in the defer function  in `createSandboxFromConfig()` function.